index.bs

```diff
@@ -9827,58 +9827,66 @@ Methods</h5>
+- <dfn for="processor construction data">node</dfn> which is a reference to
+	the {{AudioWorkletNode}} created.
+- <dfn for="processor construction data">options</dfn> which a serialized
+	{{AudioWorkletNodeOptions}} given to the {{AudioWorkletNode}}'s
+	{{AudioWorkletNode()|constructor}}.
+- <dfn for="processor construction data">port</dfn> which a serialized
+	{{MessagePort}} paired with the {{AudioWorkletNode}}'s
+	{{AudioWorkletNode/port}}.
 
 Upon the arrival of the transferred data on the {{AudioWorkletGlobalScope}},
-the <a>rendering thread</a> will trigger the algorithm below:
+the <a>rendering thread</a> will invoke the algorithm below:
 
 	<div id="invoking-processor-constructor" algorithm="invoking processor constructor">
-		Upon the arrival of <a>processor construction data</a>
-		<var>constructionData</var>, perform the following steps. If
-		any step throws an exception, abort the remaining steps and
-		<a>queue a task</a> to fire an
-		<a href="https://www.w3.org/TR/html50/webappapis.html#the-errorevent-interface">
-		ErrorEvent</a> named <code>processorerror</code> at the
-		{{AudioWorkletNode}} on the <a>control thread</a> with the
-		relevant information about the error.
-
-		1. Let <var>name</var> be the [=nodeName=] member of
-			<var>constructionData</var>.
-
-		1. Let <var>options</var> be the [=nodeOptions=] member of
-			<var>constructionData</var>.
-
-		1. Let <var>port</var> be the [=nodePort=] member of
-			<var>constructionData</var>.
-
+		1. Let <var>constructionData</var> be the
+			[=processor construction data=] transferred from the
+			[=control thread=].
+
+		1. Let <var>processorName</var> be <var>constructionData</var>'s
+			[=processor construction data/name=].
+
+		1. Let <var>nodeReference</var> be <var>constructionData</var>'s
+			[=processor construction data/node=].
+
+		1. Let <var>serializedPort</var> be <var>constructionData</var>'s
+			[=processor construction data/port=].
+		
+		1. Let <var>serializedOptions</var> be
+			<var>constructionData</var>'s
+			[=processor construction data/options=].
+		
+		1. Let <var>deserializedPort</var> be the result of
+			[$StructuredDeserialize$](<var>serializedPort</var>,
+			the current Realm).
+
+		1. Let <var>deserializedOptions</var> be the result of
+			[$StructuredDeserialize$](<var>serializedOptions</var>,
+			the current Realm).
+		
 		1. Let <var>processorCtor</var> be the result of looking
-			up <var>name</var> on the {{AudioWorkletGlobalScope}}'s
+			up  <var>processorName</var> on the
+			{{AudioWorkletGlobalScope}}'s
 			<a>node name to processor constructor map</a>.
-
-		1. Perform Construct(<var>processorCtor</var>,
-			« <var>options</var> ») with the given <var>port</var>.
+		
+		1. Make <dfn><var>nodeReference</var></dfn>,
+			<dfn><var>deserializedPort</var></dfn>, and
+			<var>deserializedOptions</var> available
+			for the execution of <var>processorCtor</var>.
+
+		1. Invoke the <var>processorCtor</var> with the argument of
```

Comment:
Perhaps it's better to say `Perform Construct(processorCtor, deserializedOptions)` like the step 9 of https://heycam.github.io/webidl/#construct-a-callback-function.

I am not sure `deserializedOptions` is a valid ES value. Probably it needs a conversion?

Reply:
Why would we prefer `Construct` over "construct a callback function", especially since what we have is in fact a callback function?  I mean, we _can_ do a manual `Construct` but then we will need to also manually duplicate a lot of the bookkeeping "construct a callback function" does for us, no?

`deserializedOptions` is the return value of a `StructuredDeserialize` call, right?  That would be an ES value, as defined at https://html.spec.whatwg.org/multipage/structured-data.html#structureddeserialize

Reply:
I don't have any preference here. I am merely focusing two things: your suggestion and minimizing changes.

> I mean, we can do a manual Construct but then we will need to also manually duplicate a lot of the bookkeeping "construct a callback function" does for us, no?

It sounds like you're constructing a callback function, but not executing the constructor itself. I was mistaken. If this is the correct way of performing the constructor (processorCtor), then I am all for it.

Also the algorithm has some interesting bits:
"Let relevant settings be realm’s settings object."

Is this "setting object" similar to "processor construction data"? Then this algorithm might be a solution for our situation.

> deserializedOptions is the return value of a StructuredDeserialize call, right? That would be an ES value

Understood.

Reply:
> If this is the correct way of performing the constructor

Yes, it is.

> Is this "setting object" similar to "processor construction data"?

No, it's the thing defined at https://html.spec.whatwg.org/multipage/webappapis.html#settings-object

Reply:
ACK